go/pkg/pwdb/migrations.go

```diff
@@ -1,54 +1,18 @@
 package pwdb
 
 import (
-	"github.com/jinzhu/gorm"
-	"gopkg.in/gormigrate.v1"
+	"gorm.io/gorm"
 	"pathwar.land/pathwar/v2/go/pkg/errcode"
 )
 
-func migrate(db *gorm.DB, opts Opts) error {
-	migrateOpts := gormigrate.DefaultOptions
-	migrateOpts.UseTransaction = true
-	m := gormigrate.New(db, migrateOpts, []*gormigrate.Migration{})
-
-	// only called on fresh database
-	m.InitSchema(func(tx *gorm.DB) error {
-		tx.Set("gorm:table_options", "charset=utf8mb4")
-		err := tx.AutoMigrate(All()...).Error
-		if err != nil {
-			tx.Rollback()
-			return errcode.ErrDBAutoMigrate.Wrap(err)
-		}
-
-		if !opts.skipFK {
-			for _, fk := range ForeignKeys() {
-				e := ByName(fk[0])
-				if err := tx.Model(e).AddForeignKey(fk[1], fk[2], "RESTRICT", "RESTRICT").Error; err != nil {
-					tx.Rollback()
-					return errcode.ErrDBAddForeignKey.Wrap(err)
-				}
-			}
-		}
```

Comment:
are those foreign keys automatically created now or are we losing them?

Reply:
They are automatically created and constraints are customizable directly on proto file :)